src/toil/test/batchSystems/batchSystemTest.py

```diff
@@ -415,6 +418,29 @@ def testIgnoreNode(self):
         # Make sure job is NOT running
         self.assertEqual(set(runningJobIDs), set({}))
 
+def write_temp_file(s: str) -> str:
+    """
+    Dump a string into a temp file and return its path.
+    """
+    fd, path = tempfile.mkstemp()
+    try:
```

Comment:
Please use pytest for temp file creation, so that they get deleted after the tests

Reply:
I can't use `pytest`'s fixtures to manage temporary files here; they aren't available in `unittest.TestCase` tests like all the Toil tests are.

But, Toil already has `_createTempDir()` and `_getScriptSource()` methods on its test case base class, which is going to be the Right Way to do those things, so I am using those now.